### PR TITLE
knative: add an extension for installing knative and deploying knative-based resources

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/knative:

6751b2e07826c30b75d03ddbcc92e28dcdcf9190 (2021-11-17 18:49:17 -0500)
knative: add an extension for installing knative and deploying knative-based resources

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
i've wanted this for so long!!! and we finally have the primitives to support it well!

**Review by nicksieger — COMMENTED** [review]
Do we typically put version guards in extensions? This one requires Tilt >= 0.23 right?

File: knative/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  k8s_custom_deploy(
+    'knative-crds',
+    deps=[],
+    apply_cmd="""
+set -e
+kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.0.0/serving-crds.yaml -o yaml
+kubectl wait --for=condition=Established crd -l=knative.dev/crd-install=true 1>&2
+""",
+    delete_cmd="""
```
Comment (milas) [inline knative/Tiltfile:22]: I could see it being useful to have an arg for `knative_install` to control down policy for for the CRDs/knative core, but not critical by any means

**Review by milas — APPROVED** [review]
This is awesome!

File: knative/Tiltfile (hunk 1 above)
Comment (nicks) [reply knative/Tiltfile:22]: yahhhh... i could see a couple different 'down' approaches:
- delete the whole knative namespace
- delete nothing at all!

but also thinks this dovetails a lot with the work we're doing around resource subsets

**Review by nicks — COMMENTED** [review]
re: version constraints - added!

**Pull request merged**